appserver/orb/orb-iiop/src/main/java/org/glassfish/enterprise/iiop/impl/IiopFolbGmsClient.java

```diff
@@ -352,12 +357,17 @@ private ClusterInstanceInfo getClusterInstanceInfo( Server server,
         final int weight = Integer.parseInt( server.getLbWeight() ) ;
         fineLog( "getClusterInstanceInfo: weight {0}", weight ) ;
 
+        final IiopService iservice = config.getExtensionByType(IiopService.class) ;
+        fineLog( "getClusterInstanceInfo: iservice {0}", iservice ) ;
+        
         final String nodeName = server.getNodeRef() ;
         String hostName = nodeName ;
         if (nodes != null) {
             Node node = nodes.getNode( nodeName ) ;
             if (node != null) {
-                if (node.isLocal()) {
+                if (node.isLocal() && (!Boolean.getBoolean(USE_NODE_HOST_FOR_LOCAL_NODE_SYSTEM_PROPERTY) 
+                        || !Boolean.parseBoolean(iservice.getOrb().getPropertyValue(
```

Comment:
A condition of type "NOT condition1 OR NOT condition2" always smells - it should be "NOT condition1 AND NOT condition2", or changed to "NOT (condition1 OR condition2)"